### PR TITLE
[battery] Allow to control charging hysteresis. JB#58372

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

**Review by vigejolla — APPROVED** [review]
LGTM

**Comment by martyone** [comment]
Refactored to use MCE's settings API. The diff to look at is https://github.com/sailfishos/nemo-qml-plugin-systemsettings/compare/ff4f4e3693f0a1f3661413451a374b4fc8c578fe..47a3e7e82231a0e6680753271c58b556660aec58.

**Comment by martyone** [comment]
Refactored the write side too, fixed D-Bus signal handling.

**Comment by martyone** [comment]
Updated after dropping change to enum ordering on mce side.

File: src/batterystatus.cpp [hunk 1]
```diff
@@ -39,12 +39,26 @@
 #include <mce/dbus-names.h>
 #include <mce/mode-names.h>
 
+namespace {
+const QString MceSettingsChargingMode = "/system/osso/dsm/charging/charging_mode";
+const QString MceSettingsChargingLimitEnable = "/system/osso/dsm/charging/limit_enable";
+const QString MceSettingsChargingLimitDisable = "/system/osso/dsm/charging/limit_disable";
```
Comment (rainemak) [inline src/batterystatus.cpp:45]:
```
const auto MceSettingsChargingLimitDisable = QStringLiteral("/system/osso/dsm/charging/limit_disable");
```
I tend to use QStringLiterals for this kind of constants. Woboq guys wrote a good article back in time: https://woboq.com/blog/qstringliteral.html

Comment (martyone) [reply src/batterystatus.cpp]: Fixed

**Review by rainemak — APPROVED** [review]

File: src/batterystatus.cpp [hunk 2]
```diff
@@ -113,6 +167,37 @@ void BatteryStatusPrivate::registerSignals()
...
+
+    return BatteryStatus::EnableCharging;
+}
+
+int BatteryStatusPrivate::chargingModeToInt(BatteryStatus::ChargingMode mode)
+{
+    switch (mode) {
+    default:
+    case BatteryStatus::EnableCharging:
+        return MceChargingModeEnable;
```
Comment (spiiroin) [inline src/batterystatus.cpp:191]: consistency nit: return from default: here vs. return from outside switch in the above function. Both ways are just fine, but now there are two very similar methods that look different. Feel free to ignore ;-)

**Review by spiiroin — COMMENTED** [review]
No real issues with what is already here, but that charge-full-once and/or charge-full-after-connect discussion would affect what is needed here -> I'd like to see that resolved 1st.

**Pull request merged**